### ISSUE TITLE
[BUG] Issue with Next Waypoint and Clear Waypoint

### ISSUE THREAD

**Issue by hihello2017** [flybywiresim/a32nx#2694]
**Mod Version**
{
  "built": "2020-12-24T20:05:38+00:00",
  "ref": "",
  "sha": "",
  "actor": "hotshotp <70779662+hotshotp@users.noreply.github.com>",
  "event_name": "manual"
}

**Describe the bug**
Clearing a waypoint results in clearing the previous waypoint instead of to be cleared waypoint
If B should be the next waypoint of A, then B is added before A instead of next A

**To Reproduce**
1. Insert LTAY ATGIT UT30 SIVKU UL610 ETAMP
2. Clear a random waypoint (press CLR in MCDU)
3. Add a random waypoint OKIPI after ETAMP 

**Expected behavior**
2. The selected waypoint shall be cleared
3.  OKIPI shall be after ETAMP

**Actual behavior**
2. The previous waypoint is cleared instad of the to be cleared waypoint
3. OKIPI is added before ETAMP instead of after ETAMP

**Comment by theersink** [comment]
> **Mod Version**
> {
> "built": "2020-12-24T20:05:38+00:00",
> "ref": "",
> "sha": "",
> "actor": "hotshotp [70779662+hotshotp@users.noreply.github.com](mailto:70779662+hotshotp@users.noreply.github.com)",
> "event_name": "manual"
> }
> 
> **Describe the bug**
> Clearing a waypoint results in clearing the previous waypoint instead of to be cleared waypoint
> If B should be the next waypoint of A, then B is added before A instead of next A
> 
> **To Reproduce**
> 
> 1. Insert LTAY ATGIT UT30 SIVKU UL610 ETAMP
> 2. Clear a random waypoint (press CLR in MCDU)
> 3. Add a random waypoint OKIPI after ETAMP
> 
> **Expected behavior**
> 2. The selected waypoint shall be cleared
> 3. OKIPI shall be after ETAMP
> 
> **Actual behavior**
> 2. The previous waypoint is cleared instad of the to be cleared waypoint
> 3. OKIPI is added before ETAMP instead of after ETAMP

Same issue here. Make for manually editing flightplans impossible.

**Comment by hihello2017** [comment]
Seems to be solved with the newest developer version.

**Comment by hihello2017** [comment]
Not working again after the last update

**Comment by Benjozork** [comment]
Hello, this problem is random and the dev version does not affect it's appearance. What you're experiencing is just the MSFS navigation system being random.

**Comment by tareksabet** [comment]
> 3. OKIPI is added before ETAMP instead of after ETAMP

that's a normal behavior it should always enter the Waypoint before even IRL, so like if I typed OKIPI then I pressed on ETAMP it should insert above it so when trying to add a waypoint add what's after it. 
Also I tried to erase a waypoint it acted as expected clearing the selected Waypoint.

Just make sure you have the latest master version. 

Thanks for reporting this issue, It can't be reproduced.
